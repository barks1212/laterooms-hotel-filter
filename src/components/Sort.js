import React from 'react';
import PT from 'prop-types';

const Sort = (props) => (
  <section className="sort is-paddingless is-marginless">
    <span className="is-paddingless has-text-weight-bold"> Sort by </span>
    <button className="button is-small is-paddingless is-marginless is-text" id="sortButton" onClick={() => props.sorter('descending', props.hotelList)}>Descending</button>
    |
    <button className="button is-small is-paddingless is-marginless is-text" id="sortButton" onClick={() => props.sorter('ascending', props.hotelList)}>Ascending</button>
  </section>
);

Sort.propTypes = {
  sort: PT.string.isRequired,
  hotelList: PT.array.isRequired,
  sorter: PT.func.isRequired
};

export default Sort;