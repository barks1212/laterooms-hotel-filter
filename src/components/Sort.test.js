import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import React from 'react';
import renderer from 'react-test-renderer';
import Sort from './Sort';

describe('<Sort>', () => {
  it('renders as expected without crashing and matches snapshot', () => {
    const hotels = [
      {
        Name: 'Yacht Harbour',
        StarRating: 5,
        Facilities: [
          'car park',
          'spa',
          'pool',
          'gym',
          'full board',
          'bar',
          'restaurant'
        ],
        Image: 'https://s-ec.bstatic.com/images/hotel/max1024x768/308/30801707.jpg'
      },
      {
        Name: 'The Adriana',
        StarRating: 3,
        Facilities: [
          'car park',
          'pool',
          'bar',
          'half board'
        ],
        'Image': 'https://t-ec.bstatic.com/images/hotel/max1024x768/259/25947064.jpg'
      },
    ];


    const tree = renderer.create(<Sort sort={{ temp: false }} hotelList={hotels} sorter={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});