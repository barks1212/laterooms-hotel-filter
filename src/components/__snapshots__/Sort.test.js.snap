// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort> renders as expected without crashing and matches snapshot 1`] = `
<section
  className="sort is-paddingless is-marginless"
>
  <span
    className="is-paddingless has-text-weight-bold"
  >
     Sort by 
  </span>
  <button
    className="button is-small is-paddingless is-marginless is-text"
    id="sortButton"
  >
    Descending
  </button>
  |
  <button
    className="button is-small is-paddingless is-marginless is-text"
    id="sortButton"
  >
    Ascending
  </button>
</section>
`;
